Bee Attack

PaulTashima opened the pull request:
Added bee attack:
- Animations created for idle, transformation, angry, and death.
- Bee begins idle with movement in the animator behavior script. In proximity to the player, it transitions to Angry.
- While angry it engages the base FollowBossController Move routine.
- Bee is not removed on collision with player (unlike Bubble).

@rhonig22 This was a dirty update. Let me know if you have thoughts on cleaning it up.

buzzbuzz55 committed:
Bee Attack

Added bee attack:
- Animations created for idle, transformation, angry, and death.
- Bee begins idle with movement in the animator behavior script. In proximity to the player, it transitions to Angry.
- While angry it engages the base FollowBossController Move routine.
- Bee is not removed on collision with player (unlike Bubble).

rhonig22 commented on Assets/Scripts/Enemy/FrogBoss/FrogBossController.cs:
```diff
@@ -10,6 +10,7 @@ public class FrogBossController : BaseBossController
     private readonly int _bubbleStormCount = 5;
     private readonly float _bubbleDelay = .4f;
     [SerializeField] private GameObject _bubble;
+    [SerializeField] private GameObject _bee;
```
It looks like you haven't dropped the Bee prefab into this field on the frog object.

rhonig22 commented on Assets/Scripts/Enemy/FrogBoss/FrogBossController.cs:
```diff
@@ -122,6 +124,7 @@ private void CreateBubbles(int count)
         var playerDistance = (Vector3)_playerLocation - transform.position;
         Vector3 spawnPosition = transform.position;
         Vector3 offset = Vector3.zero;
+        // Later: one of these should be playerDistance.y 
```
Woops! The second value should indeed be Y

rhonig22 commented on Assets/Scripts/Enemy/Minions/BeeIdleBehavior.cs:
```diff
@@ -0,0 +1,52 @@
+    {
+        
+        // Check if the bee should be angry
+        _playerLocation = GameObject.FindWithTag("Player").transform.position;
```
Store the player object in a var in onStateEnter and then just grab its position here, so that you don't have to search for the player object every frame.

rhonig22 commented on Assets/Scripts/Enemy/Minions/BeeIdleBehavior.cs:
```diff
@@ -0,0 +1,52 @@
+
+        // Check if destination was reached, then select new destination
+        var destinationDistance = (Vector3)_controller.transform.position - _endPosition;
+        if(Mathf.Abs(destinationDistance.magnitude) < 1)
```
Maybe break this logic out to a helper method called _setNewTargetPosition

rhonig22 commented on Assets/Scripts/Enemy/Minions/BeeController.cs:
```diff
@@ -0,0 +1,38 @@
+public class BeeController : FollowBossController
+{
+    protected int _beeHealth = 20;
+    public bool isAngry = false;
```
public var should have the first letter capitalized: IsAngry

buzzbuzz55 committed:
Comment resolution

All comments resolved:
- playerDistance.x updated to playerDistance.y
- _player GameObject created in BeeIdleBehavior for faster reference
- _setNewTargetPosition() created
- IsAngry capitalized.
